torch/functional.py

```diff
@@ -1589,3 +1590,49 @@ def _lu_no_infos(A, pivot=True, get_infos=False, out=None):
 
 def align_tensors(*tensors):
     raise RuntimeError('`align_tensors` not yet implemented.')
+
+def result_type(*arrays_and_dtypes: Union[Tensor, torch.dtype, bool, int, float, complex]) -> torch.dtype:
+    """result_type(*arrays_and_dtypes: Union[Tensor, dtype, bool, int, float, complex]) -> dtype
```

Comment:
I believe the python array API standard mentions leaving out type annotations from signature and only including them in the parameter list (https://data-apis.org/array-api/latest/API_specification/data_type_functions.html?highlight=result_type#result-type-arrays-and-dtypes).